jsonrpc/server.go

```diff
@@ -359,20 +377,33 @@ func (s *Server) handleBatchRequest(ctx context.Context, batchReq []json.RawMess
 					resp.ID = req.ID
 				}
 			}
-			// for notification request response is nil
+			// for notification request response is nil and header is irrelevant for now
 			if resp != nil {
-				addResponse(resp)
+				addResponse(resp, header)
 			}
 		})
 	}
 
 	wg.Wait()
+
+	// merge headers
+	finalHeaders := http.Header{}
+	for _, header := range headers {
+		for k, v := range header {
+			for _, e := range v {
+				finalHeaders.Add(k, e)
+			}
+		}
+	}
```

Comment:
Just making a note for myself to come back to this, I don't fully understand what's happening here

Reply:
so whats happening here is 
the headers returned from all the calls to the different handlers (batch calls) are being merged into one.

result would look like this 
```go

// header returned by first call
{
"Header 1":  "value from call 1",
"Header 3":  "value from call 1"
}

// header returned by second call 
{
"Header 2":  "value from call 2",
"Header 3":  "value from call 2"
}


// final header on combined result
{
 "Header 1": "value from call 1",
 "Header 2": "value from call 1, value from call 2",
 "Header 3": "value from call 2"
}
```


its a work around seeing that to support single return headers for batch requests as it is currently implemented **may** require a complete refactor to remove its dependency on single request calls.

Reply:
yea!

not so related Side note: an additional question the question is when does it make sense to aggregate the  different headers values in to one value (like total execution steps)